### ISSUE TITLE
Editing a gallery shows all images from the media library as selected

### ISSUE THREAD

**Issue by florianbrinkmann** [WordPress/gutenberg#9072]
**Describe the bug**

After creating a gallery block and uploading an image to it – via drag & drop or the upload button – all images of the media library are displayed as selected in the edit gallery view. I am using Gutenberg 3.5.

**To Reproduce**
1. Insert a gallery block.
2. Upload one or more images to it via drag & drop or the upload button.
3. Click the Edit Gallery button
4. See all images of the library as selected.

**Expected behavior**

Just the images from the gallery should be set as selected.

**Screenshots**

![gutenberg-gallery-issue](https://user-images.githubusercontent.com/2310924/44256482-93084c00-a209-11e8-9625-d405a56d077f.gif)

**Desktop (please complete the following information):**
 - Windows 10
 - Firefox Nightly
 - 63.0a1 (2018-08-16)

**Comment by designsimply** [comment]
Good find! Tested and confirmed with WordPress 4.9.8 and Gutenberg 3.6.2 using Firefox 61.0.2 and Chrome 68.0.3440.106 on macOS 10.13.6. ([1m29s](https://cloudup.com/cyvQlzzEL9I))

![all-images-should-not-be-selected-by-default-in-the-edit-gallery-screen](https://user-images.githubusercontent.com/1119271/44285632-59443f00-a223-11e8-997e-87996c8e048d.png)

**Comment by jeherve** [comment]
This is still happening today. I was able to reproduce this just now. 👍

**Comment by paaljoachim** [comment]
Btw - An update for GB version 5.0 Rc-1

Uploading media:  Still having the same gallery block selected click the pencil icon. All the images in the media library shows up. I clicked cancel. Deselected the gallery block and then selected it again. Then only 4 images showed up in the Edit Gallery view.

**Comment by antpb** [comment]
I started digging into this. Wanted to document some of my initial findings as I find a solution.
I'm noticing we check for  `images.length` to render the edit toolbar. Then we set the value of the edit button to `images.map( ( img ) => img.id )` 
src: 
https://github.com/WordPress/gutenberg/blob/d9b232d1d15db418d22db89c53626542a8efd654/packages/block-library/src/gallery/edit.js#L198-L216


The images object updates as each photo dropped is uploaded. I'm thinking that the initial `value` is mapping through an empty `images` attribute and is holding that value.  To test this I added a log right after the condition. You'll notice that the object starts as a blank array, then starts filling in the ids as they upload. I think that when the array is initially set, it passes the `images.length` check and assigns the `value` field to a blank array. For whatever reason, the value field of the `MediaUpload` component is not updating with the updates to the `images` attribute. Going to continue investigating. :)

<img width="820" alt="screen shot 2019-02-22 at 11 35 24 am" src="https://user-images.githubusercontent.com/8985705/53259934-f9c56100-3695-11e9-869f-c728c7e797e4.png">
